addons/website_sale/controllers/main.py

```diff
@@ -1175,14 +1175,18 @@ def values_postprocess(self, order, mode, values, errors, error_msg):
                 new_values['parent_id'] = commercial_partner.id
         return new_values, errors, error_msg
 
-    def should_skip_delivery_step(self, delivery_methods):
-        """ Check if the delivery step should be skipped based on the available delivery methods.
+    def should_skip_delivery_step(self, order_sudo, delivery_methods):
+        """ Check if the delivery step should be skipped based on the available delivery methods and
+        the order_sudo.carrier_id.
+        If a delivery method is not set on the order then get_rate failed or no dms are available
```

Comment:
Blank line between the docstring "summary" and body.
```suggestion

        If a delivery method is not set on the order then get_rate failed or no dms are available
```